todo/src/reducers/index.js

```diff
@@ -0,0 +1,44 @@
+import { ADD_TODO, TOGGLE_TODO, REMOVE_TODO } from '../actions'
+
+let id = 0
+
+const setId = action => {
+  return id++
+}
+
+const initialState = [
+    {
+      value: 'Walk the dog',
+      completed: false,
+      id: setId()
+    }
+]
+
+const reducer = (state = initialState, action) => {
+  switch(action.type) {
+
+    case ADD_TODO:
+      return [
+          ...state,
+          {
+            value: action.payload,
+            completed: false,
+            id: setId()
+          }
+
+      ]
+    case TOGGLE_TODO:
+
+      let changedTodo = state.map(todo => (todo.id === action.payload) ? {...todo, completed: !todo.completed} : todo)
```

Comment:
Super! Very clean and concise. 

Note you can also return changedTodo straight away when you do the `.map` like:
```
  return state.map( ....rest of code here.... )
```
this way you left less variable foot print.